hwmgr-plugins/api/openapi.yaml

```diff
@@ -0,0 +1,744 @@
+  schemas:
+    APIVersion:
+      description: |
+        Information about a version of the API.
+      type: object
+      properties:
+        version:
+          type: string
+          example: "1.0.0"
+
+    APIVersions:
+      description: |
+        Information about a list of versions of the API.
+      type: object
+      properties:
+        uriPrefix:
+          type: string
+          example: "/hardware-manager/provisioning/v1"
+        apiVersions:
+          type: array
+          items:
+            $ref: "#/components/schemas/APIVersion"
+          example:
+            - version: "1.0.0"
+
+    ProblemDetails:
+      type: object
+      properties:
+        type:
+          type: string
+          format: uri
+          description: |
+            A URI reference according to IETF RFC 3986 [3] that identifies the problem type. It is encouraged that the URI provides
+            human-readable documentation for the problem (e.g. usingHTML) when dereferenced. When this member is not present,
+            its value is assumed to be "about:blank".
+          example: 'https://problem.description.com'
+        title:
+          type: string
+          description: |
+            A short, human-readable summary of the problem type. It should not change from occurrence to occurrence of the problem,
+            except for purposes of localization. If type is given and other than "about:blank", this attribute shall also be provided.
+        status:
+          type: integer
+          format: int
+          description: The HTTP status code for this occurrence of the problem.
+        detail:
+          type: string
+          description: A human-readable explanation specific to this occurrence of the problem.
+          example: Connection to database timed out
+        instance:
+          type: string
+          format: uri
+          description: |
+            A URI reference that identifies the specific occurrence of the problem.
+            It may yield further information if dereferenced.
+        additionalAttributes:
+          type: object
+          additionalProperties:
+            type: string
+          description: Any number of additional attributes, as defined in a specification or by an implementation.
+      required:
+        - status
+        - detail
+
+    NodeAllocationRequest:
+      description: |
+        Information about a NodeAllocationRequest resource.    
+      type: object
+      properties:
+        hardwarePluginRef:
+          type: string
+          description: |
+            Reference to the registered Hardware Plugin.        
+        nodeGroup:
+          type: array
+          items:
+            $ref: "#/components/schemas/NodeGroup"
+          description: |
+            List of node groups which this resource is based on.
+        site:
+          type: string
+          description: |
+            Site identifier associated with the NodeAllocationRequest.
+        extensions:
```

Comment:
How do we envision using `extensions`?   If we have extensions then it means we may have vendor specific functionality on our side.  Is that something we want?

cc: @browsell

Reply:
+ @donpenney

Reply:
In this model I don't see the point of extensions as the IMS would not be able to consume it.

Reply:
I will remove the `extensions` and we can add it back if later required.